app/src/main/java/com/infomaniak/mail/data/models/message/Message.kt

```diff
@@ -109,15 +111,30 @@ class Message : RealmObject {
         return this
     }
 
-    fun setDraftId(draftUuid: String?) {
-        MailRealm.mailboxContent.writeBlocking { getLatestMessage(uid)?.draftUuid = draftUuid }
-    }
-
     fun getDkimStatus(): MessageDKIM? = enumValueOfOrNull<MessageDKIM>(dkimStatus)
 
     enum class MessageDKIM(val value: String?) {
         VALID(null),
         NOT_VALID("not_valid"),
         NOT_SIGNED("not_signed"),
     }
+
+    companion object {
+        fun from(draft: Draft) = Message().apply {
+            uid = draft.parentMessageUid.ifEmpty { draft.uuid }
+            draftUuid = draft.uuid
+            subject = draft.subject
+            folder = API_DRAFT_FOLDER_NAME
+            folderId = getDraftsFolder()?.id.toString()
+            from = draft.from
+            to = draft.to ?: realmListOf()
+            cc = draft.cc ?: realmListOf()
+            bcc = draft.bcc ?: realmListOf()
+            replyTo = draft.replyTo
+            isDraft = true
+            attachments = draft.attachments
+            hasAttachments = draft.attachments.isNotEmpty()
+            date = draft.date
+        }
+    }
```

Comment:
to see if we can simplify it with an interface

Reply:
It will be done later in a different PR